app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryBranchesViewModel.kt

```diff
@@ -0,0 +1,68 @@
+package com.bottlerocketstudios.brarchitecture.ui.repository
+
+import com.bottlerocketstudios.brarchitecture.R
+import com.bottlerocketstudios.brarchitecture.domain.models.Branch
+import com.bottlerocketstudios.brarchitecture.domain.repositories.BitbucketRepository
+import com.bottlerocketstudios.brarchitecture.ui.BaseViewModel
+import com.bottlerocketstudios.compose.repository.RepositoryBranchItemUiModel
+import com.bottlerocketstudios.compose.util.formattedUpdateTime
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import org.koin.core.component.inject
+import timber.log.Timber
+import java.time.Clock
+
+class RepositoryBranchesViewModel : BaseViewModel() {
+    // DI
+    private val repo: BitbucketRepository by inject()
+    private val clock by inject<Clock>()
+
+    // State
+    private val srcBranches = MutableStateFlow<List<Branch>>(emptyList())
+    val currentRepoName = MutableStateFlow("")
+
+    // Internal state
+    val repos = repo.repos.groundState(emptyList())
+
+    init {
+        launchIO {
+            currentRepoName.collect { repoName ->
+                getRepoBranches(repoName)
+            }
+        }
+    }
+
+    // UI
+    val path: StateFlow<String> = MutableStateFlow("")
+    val itemCount: StateFlow<Int> = srcBranches
+        .map { it.size }
+        .groundState(0)
+    val uiModels: StateFlow<List<RepositoryBranchItemUiModel>> = srcBranches
+        .map { branches ->
+            branches.map { branch ->
+                RepositoryBranchItemUiModel(
+                    name = branch.name,
+                    timeSinceCreated = branch.date.formattedUpdateTime(clock),
+                    status = "#1(OPEN)"
```

Comment:
@LuisRocketRangel Can we derive this data from something else? Is this data overwritten at some point?

* If we need to keep it and it is actually shown in the UI at some point, let's move it to `strings.xml` and probably change `RepositoryBranchItemUiModel.status` from a `String` to `StringIdHelper`
* If not, replace this with the empty string.

Reply:
The status is dynamic but the branch call doesn't contain this data. Replaced with the StringIDHelper